tensorboard/compat/tensorflow_stub/pywrap_tensorflow.py

```diff
@@ -220,25 +230,59 @@ def GetNext(self):
         # The length of the header tells us how many bytes the Event
         # string takes
         header_len = int(header[0])
-        event_str = self.file_handle.read(header_len)
+        event_str = self._read(header_len)
+        if len(event_str) < header_len:
+            raise self._truncation_error("data")
 
         event_crc_calc = masked_crc32c(event_str)
 
         # The next 4 bytes contain the crc32 of the Event string,
-        # which we check for integrity. Sometimes, the last Event
-        # has no crc32, in which case we skip.
-        crc_event_str = self.file_handle.read(4)
-        if crc_event_str:
-            crc_event = struct.unpack("I", crc_event_str)
-            if event_crc_calc != crc_event[0]:
-                raise errors.DataLossError(
-                    None,
-                    None,
-                    "{} failed event crc32 check".format(self.filename),
-                )
+        # which we check for integrity.
+        crc_event_str = self._read(4)
+        if len(crc_event_str) < 4:
+            raise self._truncation_error("data crc")
+        crc_event = struct.unpack("I", crc_event_str)
+        if event_crc_calc != crc_event[0]:
+            raise errors.DataLossError(
+                None, None, "{} failed event crc32 check".format(self.filename),
+            )
```

Comment:
We’re raising the same error for a CRC failure as a truncated stream.
How does upstream code know to retry on the latter but skip this record
on the former? From a cursory read it looks to me like this would
infinitely loop raising `DataLossError`s once we find a corrupt record.

Reply:
For better or for worse, this is the same behavior that the TF C++ RecordReader has always had:
https://github.com/tensorflow/tensorflow/blob/a34091e538540aad57a7a941575538944f38db24/tensorflow/core/lib/io/record_reader.cc#L99
https://github.com/tensorflow/tensorflow/blob/a34091e538540aad57a7a941575538944f38db24/tensorflow/core/lib/io/record_reader.cc#L105

So I'm not too inclined to try to fix it here given that we can't really fix it in the normal path other than by attempting to parse the string error message on the resulting status, which seems too brittle.  I think corrupted checksums are infrequent enough in practice that looping on them, while not ideal, is an acceptable behavior.  It's the same "infinite loop" that we would have at EOF anyway, so not really a big deal IMO.

Reply:
Wow, okay, confirmed: wrote an event file with five steps, corrupted the
third one, and observed that only the first two steps are ever loaded.
That’s not great, but you’re right that it’s not a regression.